app/src/org/commcare/views/widgets/ComboboxWidget.java

```diff
@@ -0,0 +1,123 @@
+package org.commcare.views.widgets;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.os.Build;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.AdapterView;
+
+import org.commcare.views.Combobox;
+import org.javarosa.core.model.SelectChoice;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.InvalidData;
+import org.javarosa.core.model.data.SelectOneData;
+import org.javarosa.core.model.data.helper.Selection;
+import org.javarosa.form.api.FormEntryPrompt;
+
+import java.util.Vector;
+
+/**
+ * ComboboxWidget is identical to a SpinnerWidget (a select-one widget formatted as a dropdown
+ * question), but with the added ability to filter the options in the dropdown list by typing into
+ * an EditText box.
+ *
+ * @author Aliza Stone
+ */
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+public class ComboboxWidget extends QuestionWidget {
+
+    private Vector<SelectChoice> choices;
+    private Vector<String> choiceTexts;
+    private Combobox comboBox;
+
+    public ComboboxWidget(Context context, FormEntryPrompt prompt, boolean permissive) {
+        super(context, prompt);
+        initChoices(prompt);
+        comboBox = new Combobox(context, choiceTexts, true, permissive, mQuestionFontSize);
+        addView(comboBox);
+
+        comboBox.setEnabled(!prompt.isReadOnly());
+        comboBox.setFocusable(!prompt.isReadOnly());
+        addListeners();
+        fillInPreviousAnswer(prompt);
+    }
+
+    private void initChoices(FormEntryPrompt prompt) {
+        choices = prompt.getSelectChoices();
+        choiceTexts = new Vector<>();
+        for (int i = 0; i < choices.size(); i++) {
+            choiceTexts.add(prompt.getSelectChoiceText(choices.get(i)));
+        }
+    }
+
+    private void addListeners() {
+        comboBox.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                widgetEntryChanged();
+            }
+        });
+
+        comboBox.addTextChangedListener(new TextWatcher() {
+            @Override
```

Comment:
Does mean there's no signal for "the user typing out a correct entry, not clicking on the entry in the list, and then clicking on the next question" which results in the value being set? Clicking the ">" or swiping would do it, since that triggers a "get answer", but I wonder if, say, losing focus should also attempt to set it?